pyodide-build/pyodide_build/pypabuild.py

```diff
@@ -0,0 +1,108 @@
+import contextlib
+import os
+import sys
+import traceback
+from itertools import chain
+from pathlib import Path
+from typing import Mapping
+
+from build import BuildBackendException, ProjectBuilder  # type: ignore[import]
+from build.__main__ import (  # type: ignore[import]
+    _STYLES,
+    _error,
+    _handle_build_error,
+    _IsolatedEnvBuilder,
+    _ProjectBuilder,
+)
```

Comment:
These are all pretty short (longest is `_handle_build_error` at 20 lines) and don't use any other private functions from `build` so we could copy most of this into this file. We should probably use some terminal color package instead of hand implementing colors since unlike `build` we aren't part of a bootstrap.

Reply:
Rich. :D

Reply:
I feel like a lot of this is just to get the same output as build. If you used rich instead, you could probably get a really nice output too. Some reimplementation, but not too bad.

Reply:
How about we keep it like this for this PR and then make a second PR to switch to use rich to make nice terminal output and remove the dependence on these private `build` names?

Reply:
You pin to an exact build version, so it's safe, sure.